runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java

```diff
@@ -160,11 +159,23 @@ public Response createStateMachine(StateMachineDefinition stateMachineDefinition
                     append(stateMachine.getName()).
                     append(".started").toString());
         } catch (ConstraintViolationException ex) {
-            //in case of Duplicate correlation key, return http code 409 conflict
-            return Response.status(Response.Status.CONFLICT.getStatusCode()).entity(ex.getCause() != null ? ex.getCause().getMessage() : null).build();
+            //In case of Duplicate correlation key, return http code 409 conflict
+            if(ex.getCause() != null) {
+                if (ex.getCause().getMessage().toLowerCase().contains("duplicate entry")) {
+                    return Response.status(Response.Status.CONFLICT.getStatusCode()).entity(
+                            ex.getCause() != null ? ex.getCause().getMessage() : null).build();
```

Comment:
isn't this null check redundant? you already checked at line 163.

Reply:
Removed redundant check.